test/framework/clustervalidator.go

```diff
@@ -246,3 +254,26 @@ func getWorkerNodeMachineSets(ctx context.Context, vc ClusterValidatorConfig, w
 	}
 	return ms.Items, nil
 }
+
+func validateCilium(ctx context.Context, vc ClusterValidatorConfig) error {
+	clusterClient := vc.ClusterClient
+	if vc.ClusterSpec.Cluster.IsManaged() {
+		clusterClient = vc.ManagementClusterClient
+	}
+
+	yaml := vc.ClusterSpec.Cluster
+	cluster := &v1alpha1.Cluster{}
+	key := types.NamespacedName{Namespace: vc.ClusterSpec.Cluster.Namespace, Name: vc.ClusterSpec.Cluster.Name}
+	err := clusterClient.Get(ctx, key, cluster)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve cluster %s", err)
+	}
+
+	clusterCilium := cluster.Spec.ClusterNetwork.CNIConfig.Cilium.PolicyEnforcementMode
+	yamlCilium := yaml.Spec.ClusterNetwork.CNIConfig.Cilium.PolicyEnforcementMode
+	if clusterCilium != yamlCilium {
+		return fmt.Errorf("cilium config does not match. %s and %s", clusterCilium, yamlCilium)
+	}
```

Comment:
I would compare the policy in the `ClusterSpec.Cluster` to the policy referenced in `cilium-config` map instead